### ISSUE TITLE
Plan about integrating UDF to replace SDF

### ISSUE THREAD

**Issue by wangyida** [bennyguo/instant-nsr-pl#76]
Seems that works presenting the surface in unsigned distance field (UDF) such as [NeUDF*](https://arxiv.org/pdf/2303.15368.pdf) and [NeUDF](https://arxiv.org/pdf/2304.10080.pdf) is more promising for presenting slender and thin structures. Do you any plan towards structuring a practical way to implement it there in NSR?

**Comment by bennyguo** [comment]
I'm not very familiar with the differentiable rendering approaches used in these UDF papers. Do they differ much from NeRF and NeuS rendering? Currently I'm short on time to do this, but will be appreciated if you could implement a new system for things like this.

**Comment by wangyida** [comment]
@bennyguo Have geeked around the published UDF works, seems to be quite suitable for single object surface extracting, but I haven't make it work for unbounded scenes with torch_lighting. But instead my implementation with [HF-NeuS](https://proceedings.neurips.cc/paper_files/paper/2022/file/0ce8e3434c7b486bbddff9745b2a1722-Paper-Conference.pdf) seems to be reasonable, is it one of the interest for your project?
Basically I tried with projecting SDF to transmittance instead of weight as mentioned in the paper.

**Comment by bennyguo** [comment]
@wangyida Sounds great! Did you try other tricks/regularizations proposed in the HF-NeuS paper? or you just tried with the transmittance formulation?

**Comment by wangyida** [comment]
> 

Hi @bennyguo  I have only implemented the transmittance formulation [here](https://github.com/wangyida/instant-nsr-pl/blob/main/models/neus.py#L139-L146), following the edcription on page 5 in [HF-NeuS](https://openreview.net/pdf?id=UPnJuDKqOfX). I also added some cross entropy loss on weights. The high frequency network need addtional parametric model to get it described, so I didn't add it

**Comment by wangyida** [comment]
Tested with an NSR projection function as well, I assume that the projection function implemented by Benny is still NeuS instead of Instant-NSR. PSNR on Mipnerf360 is a bit higher.
```python
# Instant-NSR applies 𝜋(.) upon the estimated sddfsvalues
prev_sdf_pi = torch.sigmoid(estimated_prev_sdf * inv_s) * (1 - torch.exp(-estimated_prev_sdf * inv_s))
next_sdf_pi = torch.sigmoid(estimated_next_sdf * inv_s) * (1 - torch.exp(-estimated_next_sdf * inv_s))
prev_cdf = torch.sigmoid(prev_sdf_pi * inv_s)
next_cdf = torch.sigmoid(next_sdf_pi * inv_s)
p = prev_cdf - next_cdf
c = prev_cdf
alpha = ((p + 1e-5) / (c + 1e-5)).view(-1).clip(0.0, 1.0)
```
